### PR TITLE
#19 - Added Repository.nuspec() method

### PR DESCRIPTION
**Pull request by olegmoz**
Part of issue #19 
Getting package metadata for package identity is required to provide version details in package metadata service (`catalogEntry` part of page)

### CODE REVIEW

**Comment by paulodamaso** [comment]
@g4s8 PLease assign me

**Comment by 0crat** [comment]
This pull request [#60](https://github.com/artipie/nuget-adapter/issues/60) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/64a22bf2-62ce-4bfa-92cb-58580ca18bf1); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artpie/nuget/RepositoryTest.java [hunk 1]
```diff
@@ -156,6 +158,31 @@ void shouldFailToAddPackageWhenItAlreadyExists() throws Exception {
...
+    @Test
+    void shouldReadNuspec() {
+        final PackageIdentity identity = new PackageIdentity(
+            new PackageId("UsefulLib"),
+            new Version("2.0")
+        );
+        this.storage.save(identity.nuspecKey(), "some data".getBytes());
+        MatcherAssert.assertThat(
+            this.repository.nuspec(identity),
+            new IsNot<>(new IsNull<>())
```
Comment (paulodamaso) [inline src/test/java/com/artpie/nuget/RepositoryTest.java]: @olegmoz You should be checking the value returned in `this.repository.nuspec(identity)`, not if it's jut null or not

File: src/test/java/com/artpie/nuget/RepositoryTest.java [hunk 2]
```diff
@@ -156,6 +158,31 @@ void shouldFailToAddPackageWhenItAlreadyExists() throws Exception {
...
+    }
+
+    @Test
+    void shouldFailToReadNuspecWhenValueAbsent() {
+        final PackageIdentity identity = new PackageIdentity(
+            new PackageId("MyPack"),
+            new Version("1.0")
+        );
+        Assertions.assertThrows(
+            Exception.class,
```
Comment (paulodamaso) [inline src/test/java/com/artpie/nuget/RepositoryTest.java]: @olegmoz Here, please check the Exception type and / or message, so we can be sure that the test failed for the correct reason

**Comment by olegmoz** [comment]
@paulodamaso thank you, made changes by your comments. Please review

File: src/test/java/com/artpie/nuget/RepositoryTest.java [hunk 3]
```diff
@@ -156,6 +156,38 @@ void shouldFailToAddPackageWhenItAlreadyExists() throws Exception {
...
+            new Version("2.0")
+        );
+        final String content = String.join(
+            "",
+            "<?xml version=\"1.0\"?>",
+            "<package xmlns=\"http://schemas.microsoft.com/packaging/2013/05/nuspec.xsd\">",
+            "<metadata><id>UsefulLib</id></metadata>",
+            "</package>"
+        );
+        this.storage.save(identity.nuspecKey(), content.getBytes());
```
Comment (paulodamaso) [inline src/test/java/com/artpie/nuget/RepositoryTest.java]: @olegmoz Please inline content creation, it is used just once

Comment (olegmoz) [reply src/test/java/com/artpie/nuget/RepositoryTest.java]: @paulodamaso done

**Comment by olegmoz** [comment]
@paulodamaso ping

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR

**Pull request merged**